API/SearchService/src/util.py

```diff
@@ -71,5 +78,63 @@ def get_param_search_type(req: func.HttpRequest, name: str, required=False, defa
     return value
 
 
-def build_search_statement(keyword: str) -> str:
-    return " AND ".join([word + "~" for word in keyword.split()])
+def get_param_match_type(req: func.HttpRequest, name: str, required=False, default=MatchType.All):
+    value = get_param(req, name, required, default)
+    if isinstance(value, MatchType):
+        return value
+    try:
+        value = int(value)
+        try:
+            value = MatchType(value)
+        except ValueError:
+            raise ParameterException(f'Illegal parameter: the parameter "{name}" should be 1(All), 2(And) or 3(Or)')
+    except ValueError:
+        if isinstance(value, str):
+            if value.lower() == "all":
+                value = MatchType.All
+            elif value.lower() == "and":
+                value = MatchType.And
+            elif value.lower() == "or":
+                value = MatchType.Or
+            else:
+                raise ParameterException(f'Illegal parameter: the parameter "{name}" should be 1(All), 2(And) or 3(Or)')
+        else:
+            raise ParameterException(f'Illegal parameter: the parameter "{name}" must be the type of int or str')
+    return value
+
+
+def build_search_statement(keyword: str, match_type: MatchType) -> str:
+    if match_type == MatchType.Or:
+        return build_or_search_statement(keyword)
+    else:
+        return build_and_search_statement(keyword)
+
+
+def build_and_search_statement(keyword: str) -> str:
+    return _build_search_statement(keyword, join_word=" AND ")
+
+
+def build_or_search_statement(keyword: str) -> str:
+    return _build_search_statement(keyword, join_word=" OR ")
+
+
+def _build_search_statement(keyword: str, join_word=" AND ") -> str:
+    search_statement = []
+    exact_length = os.environ.get("EXACT_MATCH_LENGTH", 2)
+    dist1_length = os.environ.get("DISTANCE_1_MATCH_LENGTH", 4)
+    for word in keyword.split():
+        if not word:
+            continue
+        if len(word) <= exact_length:
+            word = word
+        elif len(word) <= dist1_length:
+            word = word + "~1"
+        else:
+            word = word + "~"
```

Comment:
The explicit declaration of distance will make the code more readable, so we don't pay attention to the default distance represented by `~`
```suggestion
            word = word + "~2"
```